src/server/scripts/Outland/zone_netherstorm.cpp

```diff
@@ -576,11 +582,565 @@ class spell_netherstorm_socrethars_stone : public SpellScript
+        }
+
+        if (player->GetQuestStatus(QUEST_DIMENSIONS_THE_ALL_DEVOURING) != QUEST_STATUS_INCOMPLETE)
+        {
+            SendGossipMenuFor(player, gossipText, me->GetGUID());
+            return true;
+        }
+
+        if (playersWithKillCredit.count(player->GetGUID()) == 0)
+            AddGossipItemFor(player, GOSSIP_ICON_CHAT, player->GetSession()->GetTrinityString(TRINITY_STRING_TEXT_1), 0, GOSSIP_ACTION_INFO_DEF);
+
+        SendGossipMenuFor(player, gossipText, me->GetGUID());
+        return true;
+    }
+
+    bool OnGossipSelect(Player* player, uint32 /*menuId*/, uint32 gossipListId) override
+    {
+        uint32 action = player->PlayerTalkClass->GetGossipOptionAction(gossipListId);
+        CloseGossipMenuFor(player);
+
+        if (player->GetGUID() != questStarterGUID)
+            return true;
+
+        switch (action)
+        {
+
+        case GOSSIP_ACTION_INFO_DEF:
+        {
+            playersWithKillCredit.clear();
+
+            if (Group* group = player->GetGroup())
+            {
+                for (GroupReference* ref = group->GetFirstMember(); ref; ref = ref->next())
+                {
+                    if (Player* member = ref->GetSource())
+                    {
+                        if (member->GetQuestStatus(QUEST_DIMENSIONS_THE_ALL_DEVOURING) == QUEST_STATUS_INCOMPLETE)
+                        {
+                            member->KilledMonsterCredit(NPC_SAEED_KILLCREDIT);
+                            playersWithKillCredit.insert(member->GetGUID());
+                        }
+                    }
+                }
+            }
+            else
+            {
+                player->KilledMonsterCredit(NPC_SAEED_KILLCREDIT);
+                playersWithKillCredit.insert(player->GetGUID());
+            }
+
+            Talk(NPC_SAEED_SAY_1);
+
+            Start(true, true, player->GetGUID());
+            me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+            me->SetFaction(250);
+            me->SetReactState(REACT_AGGRESSIVE);
+
+            StartEscortFollowers();
+            break;
+        }
+
+        case GOSSIP_ACTION_INFO_DEF + 1:
+        {
+            if (pausedByPlayer)
+            {
+                pausedByPlayer = false;
+                escortPaused = false;
+                SetEscortPaused(false);
+                me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+            }
+            break;
+        }
+
+        default:
+            break;
+        }
+
+        return true;
+    }
+
+    void SetData(uint32 id, uint32 value) override
+    {
+        if (id == DATA_GOSSIP_STATE)
+            currentGossipState = value;
+    }
+
+    uint32 GetData(uint32 id) const override
+    {
+        if (id == DATA_GOSSIP_STATE)
+            return currentGossipState;
+        return 0;
+    }
+
+    void StartEscortFollowers()
+    {
+        const uint32 followerEntries[] = { NPC_SAEED_FOLLOWER_1, NPC_SAEED_FOLLOWER_2, NPC_SAEED_FOLLOWER_3 };
```

Comment:
minor but: type + modifier (uint32 const), both are valid, but we prefer it here